src/lib/progress.c

```diff
@@ -127,6 +131,40 @@ void progress_report(double count, double max)
 	}
 }
 
+int progress_spinner_callback()
+{
+	if (log_get_level() == LOG_DEBUG || !isatty(fileno(stdout))) {
+		return 0;
+	}
+
+	static int index;
+
+	if (begin == 0) {
+		begin = time(NULL);
+	}
+
+	if (time(NULL) > begin) {
+		begin = time(NULL);
+		index = (index + 1) % 6;
+		info("[%c]", spinner[index]);
+		fflush(stdout);
+		index++;
```

Comment:
you are increasing index twice

Reply:
Oops..yes no wonder why i was seein only two patterns